### ISSUE TITLE
Add rule issuing warning for non-serializable fields for types marked with [Serializable]

### ISSUE THREAD

**Issue by ahoisl** [SonarSource/sonar-csharp#1211]
### Description

Add a rule similar to rule S3925 that works on types marked with [Serializable] and which issues warnings for non-serializable fields not marked with [NonSerialized], e.g. readonly fields.

### Repro steps

```csharp
[Serializable]
class Person
{
    private int _birthYear;  // compliant
    private readonly int _birthMonth;  // Noncompliant
    [NonSerialized]
    private int _notSerializedValue;  // Noncompliant
    public Person(int birthYear, int birthMonth, int notSerializedValue)
    {
        _birthYear = birthYear;
        _birthMonth = birthMonth;
        _notSerializedValue = notSerializedValue;
    }
}
```

### Expected behavior

Issues are raised for "_birthMonth" as this field is not serializable.

### Actual behavior

No issue raised for "_birthMonth".

### Known workarounds

None.

### Related information

* SonarC# Version = 6.7.1
* Visual Studio Version = VS 2017

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-2002](https://sonarsource.atlassian.net/browse/NET-2002)

[NET-2002]: https://sonarsource.atlassian.net/browse/NET-2002?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
